test-data/unit/check-parameter-specification.test

```diff
@@ -0,0 +1,19 @@
+[case testBasicParamSpec]
+from typing_extensions import ParamSpec
+P = ParamSpec('P')
+[builtins fixtures/tuple.pyi]
+
+[case testParamSpecLocations]
+from typing import Callable, List
+from typing_extensions import ParamSpec, Concatenate
+P = ParamSpec('P')
+
+x: P  # E: ParamSpec "P" is unbound
+def foo1(x: Callable[P, int]) -> Callable[P, str]:  ...
+def foo2(x: P) -> P: ...  # E: Invalid location for ParamSpec "P"
```

Comment:
What about adding a hint about where it should be used, e.g. something along these lines:
```
# N: You can use it like this: Callable[P, <return type>]
```